src/univariate/continuous/arcsine.jl

```diff
@@ -22,49 +22,67 @@ External links
 * [Arcsine distribution on Wikipedia](http://en.wikipedia.org/wiki/Arcsine_distribution)
 
 """
-immutable Arcsine <: ContinuousUnivariateDistribution
-    a::Float64
-    b::Float64
 
-    Arcsine(a::Real, b::Real) = (@check_args(Arcsine, a < b); new(a, b))
-    Arcsine(b::Real) = (@check_args(Arcsine, b > zero(b)); new(0.0, b))
-    Arcsine() = new(0.0, 1.0)
+immutable Arcsine{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+
+    Arcsine(a::T, b::T) = (@check_args(Arcsine, a < b); new(a, b))
 end
 
+Arcsine{T<:Real}(a::T, b::T) = Arcsine{T}(a, b)
+Arcsine(a::Real, b::Real) = Arcsine(promote(a, b)...)
+Arcsine(a::Integer, b::Integer) = Arcsine(Float64(a), Float64(b))
+Arcsine(b::Real) = Arcsine(0.0, b)
+Arcsine() = Arcsine(0.0, 1.0)
+
 @distr_support Arcsine d.a d.b
 
+#### Conversions
+function convert{T<:Real}(::Type{Arcsine{T}}, a::Real, b::Real)
+    Arcsine(T(a), T(b))
+end
+function convert{T <: Real, S <: Real}(::Type{Arcsine{T}}, d::Arcsine{S})
+    Arcsine(T(d.a), T(d.b))
+end
+
 ### Parameters
 
-params(d::Arcsine) = (d.a, d.b)
 location(d::Arcsine) = d.a
 scale(d::Arcsine) = d.b - d.a
+params(d::Arcsine) = (d.a, d.b)
+@inline partype{T<:Real}(d::Arcsine{T}) = T
 
 
 ### Statistics
 
-mean(d::Arcsine) = (d.a + d.b) * 0.5
+mean(d::Arcsine) = (d.a + d.b) / 2
 median(d::Arcsine) = mean(d)
 mode(d::Arcsine) = d.a
 modes(d::Arcsine) = [d.a, d.b]
 
-var(d::Arcsine) = 0.125 * abs2(d.b - d.a)
-skewness(d::Arcsine) = 0.0
-kurtosis(d::Arcsine) = -1.5
+var(d::Arcsine) = abs2(d.b - d.a) / 8
+skewness{T<:Real}(d::Arcsine{T}) = zero(T)
+kurtosis{T<:Real}(d::Arcsine{T}) = -3/2*one(T)
 
 entropy(d::Arcsine) = -0.24156447527049044469 + log(scale(d))
 
 
 ### Evaluation
 
-pdf(d::Arcsine, x::Float64) = insupport(d, x) ? 1.0 / (π * sqrt((x - d.a) * (d.b - x))) : 0.0
+function pdf(d::Arcsine, x::Real)
+    insupport(d, x) ? one(d.a) / (π * sqrt((x - d.a) * (d.b - x))) : zero(d.a)
+end
 
-logpdf(d::Arcsine, x::Float64) = insupport(d, x) ? -(logπ + 0.5 * log((x - d.a) * (d.b - x))) : -Inf
+function logpdf{T<:Real}(d::Arcsine{T}, x::Real)
+    insupport(d, x) ? -(logπ + 1/2*log((x - d.a) * (d.b - x))) : -T(Inf)
+end
 
-cdf(d::Arcsine, x::Float64) = x < d.a ? 0.0 :
-                              x > d.b ? 1.0 :
+cdf{T<:Real}(d::Arcsine{T}, x::Real) = x < d.a ? zero(T) :
+                              x > d.b ? one(T) :
                               0.636619772367581343 * asin(sqrt((x - d.a) / (d.b - d.a)))
```

Comment:
`0.636619772367581343` is a `Float64` literal. Wolfram Alpha tells me that we can write `asin(sqrt((x - d.a) / (d.b - d.a)))/π`